07-On demand scan as an API/README.md

```diff
@@ -0,0 +1,203 @@
+
+5. In the Redirect URI section, select Web and leave the URL field empty for now.
+
+6. Select Register to create the application.
+
+7. On the app Overview page, find the Application (client) ID value and record it for later.
+
+8. Create a client secret for this application to use in a subsequent step.
+
+    - From the list of pages for your client app, select Certificates & secrets, and select New client secret.
+
+    - Under Add a client secret, provide a Description. Choose when the key should expire, and select Add.
+
+9. When the secret is created, note the key value for use in a subsequent step.
+
+</br>
+
+
+[Back to top…](README.md#On-this-page)
+
+## Generate user authentication token to access subscriptions
+User has to generate authentication token in order to use APIs for any subscription.
+There are two ways to generate access tokens:
+- Using client credential flow
+- Using user authentication code flow
+
+#### Required Az Module
+``` PowerShell
+Install-Module -Name MSAL.PS -AllowClobber -Scope CurrentUser -repository PSGallery
+```
+
+### Using client credential flow
+Client crediential flow uses the client credentials(client id and client secret) to generate the token. Token will be generating against specified SPN(Service Principal Name) and **SPN must have access over the subscription** to scan or to get the control scan result.
+
+> In order to generate the token for APIs, you have to get access for the client application from WebAPI owner.
+> 1. Send the client id to WebAPI owner to grant access for client application.
+> 2. WebAPI owner will grant the access and share the scope.
+> 3. Use WebAPI scope in below command.
+
+Commands to generate the token:
+``` PowerShell
+
+$ClientSecret = '<client-secret>' | ConvertTo-SecureString -AsPlainText -Force
+
+$token = Get-MsalToken -TenantId '<tenant-id>' -ClientId '<client-id>' -ClientSecret $ClientSecret -Scopes "<WebAPI-scope>/.default"
+
+$token.AccessToken | ConvertTo-Json | Out-File '<token file path>\token.json'
+
+```
+
+[Back to top…](README.md#On-this-page)
+
+### Using user authentication code flow
+User authentication code flow uses user's crediential to generate the token. User must have access over the subscription  to scan or to get the control scan result.
+
+> In this flow, Client app registration needs "Admin consent" to access the APIs.
+> </br>
+> Grant admin consent for client app registration:
+> 1. Go to Azure Portal.
+> 2. Go to App Registration.
+> 3. Select appropriate App Registration.
+> 4. Go to 'API Permissions'.
+> 5. Click 'Grant admin consent for Microsoft'.
+
+
+Command to generate the token:
+``` PowerShell
+
+$token = Get-MsalToken -TenantId '<tenant-id>' -ClientId '<client-app-id>' -RedirectUri 'https://localhost' -Scopes '<WebAPI-scope>'
+
+```
+
+[Back to top…](README.md#On-this-page)
+
+## API to scan a subscription
+
+Use below PowerShell command to get control scan result.
```

Comment:
You should add a description of the API.
Also, the data contract explaining various scenarios of this API.

Reply:
Added description + data contract as part of 'Notes'.